backend/assimilator.py

```diff
@@ -55,7 +55,41 @@ def assimilate(data_type, data_path, data_idName, geo_path, geo_idName, out_path
                 boundary["properties"] = data_toAssimilate[properties[geo_idName]]
 
             data_assimilated = boundaries
-    # print(boundary["properties"].keys())
+
+    # If input is points
+    elif data_type == "points":
+        point_features = []
+
+        #Read in csv as list of dictionaries so each row is a dictionary with {headers : values}
+        with open(data_path, newline="", encoding="utf8") as f:
+            reader = csv.DictReader(f)
+            
+            try:
+                for row in reader:
+                    properties = []
+
+                    for key in row:
+                        try:
+                            properties.append((key, float(row[key].replace(",", "",))))
+                        except:
+                            properties.append((key, row[key]))
+                    
+                    #For each point (row in csv), detect and extract all headers (dictionary keys), store values, and add to list of dictionaries
+                    properties = {key: value for key, value in properties}
+
+                    #Form geojson format for points, dump properties as dictionaries into properties
+                    point_features.append({
+                        "type": "Feature",
+                        "geometry": {"type": "Point", "coordinates": [float(row['Lng']), float(row['Lat'])]},
+                        "properties": properties,
+                    })
+            except:
+                print("WARNING (Assimilator): Unable to assimilate row: ", row)
+            
+            print("Message (Assimilator): Found {} points for {} ".format(len(point_features),data_path))
```

Comment:
Won't go very much in the details of all the files - will have a closer look later when we will build the new data pipeline building on the **new** Borg.

However, big if clauses like these are [_code bad smells_](https://blog.codinghorror.com/code-smells/) hinting lack of proper encapsulation. If I were you I would've thought about using a function to (also) collect shared behaviours ;)

Reply:
Yes I agree- will wait for v2 to change this

Reply:
I think this sort of change is still part of v2, isn't it?